linked_list/implementation.py

```diff
@@ -7,34 +8,122 @@ class LinkedList(AbstractLinkedList):
     """
 
     def __init__(self, elements=None):
-        pass
-
+        self.start = None # initialize starting node
```

Comment:
@martinzugnoni it's intentional. XD

as of 9pm EST Tuesday, we had only finished implementing Nodes. since we knew that we'd barely have any overlapping hours to finish this assignment (given sleep and work), we agreed to comment everything (even if it seems obvious because when we were stuck in it, we were _stuck_) so that the other can follow the flow and explanation if they're a bit confused with the code.

### context
* I was really struggling with iterators + generators and asked for help. Ended up spending 2 hours with Adrian figuring stuff out in a temporary channel that he created. Some classmates saw my call for help and asked to be added to that temporary channel (including Sebastian)
* Before I went to sleep pinned a text file with the log history so that people could see everything without needing to scroll. While I was asleep, Sebastian used my notes to start implementing the LinkedList.
* I wake up early, review what Sebastian did when I was asleep, and completed the implementation. During this time, Sebastian was on the way home. We fix the merge conflict, push, and realise that it passes on Python 2 but not on Python 3. Sebastian goes to sleep.
* After work, I fix the bugs (while Sebastian is still sleeping) and it all works on Python 3.
* Sebastian wakes up and pushes to github. Everything passes.
* We both want to study our implementation when we have more time. Over-commenting means that it's less likely we'll wonder 'what were we thinking.' Definitely shouldn't be in professional code but we agreed that since we're both learning and struggling with this implementation, it was ok.